cpp/dolfinx/io/ADIOS2Writers.cpp

```diff
@@ -786,10 +788,15 @@ void FidesWriter::write(double t)
   adios2::Variable<double> var_step = define_variable<double>(*_io, "step");
   _engine->Put<double>(var_step, t);
 
-  fides_write_mesh(*_io, *_engine, *_mesh);
+  if (_reuse_mesh == false or first_write == true)
+  {
+    fides_write_mesh(*_io, *_engine, *_mesh);
+  }
   for (auto& v : _u)
     std::visit([&](const auto& u) { fides_write_data(*_io, *_engine, *u); }, v);
 
+  first_write = false;
```

Comment:
We try to avoid objects having mutable state. Can we check in the Fides file whether or not a mesh has been written?

Reply:
Done. Now it checks if the connectivity exists.